app/assets/javascripts/controllers/middleware_datasource/middleware_datasource_service.js

```diff
@@ -124,21 +124,24 @@ function MwAddDatasourceService($http, $q) {
     var deferred = $q.defer();
     var BASE_URL = '/middleware_server/jdbc_drivers';
     var parameterizedUrl = BASE_URL + '?server_id=' + serverId;
+    var serverError = 'internal_server_error';
 
     $http.get(parameterizedUrl).then(function(driverData) {
-      var transformedData = _.chain(driverData.data.data)
-        .map(function(driver) {
-          return {'id': driver.properties['Driver Name'].toUpperCase(),
-                  'label': driver.properties['Driver Name'],
-                  'moduleName': driver.properties['Module Name'],
-                  'xaDsClass': driver.properties['XA DS Class'],
-                  'driverClass': driver.properties['Driver Class']};
-      })
-      .value();
-
-      deferred.resolve(transformedData);
-    }).catch(function(errorMsg) {
-      deferred.reject(errorMsg);
+      if (driverData.data.status === serverError) {
```

Comment:
@AparnaKarve I just realized that you meant use the `API.get` for client requests. I guess I read it too quickly and assumed server-side. No reason the `API.get` pattern can't be put in right now -- agreed!